### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by Sgjzhang**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
adding codes and answer questions
## What did you learn from the changes you have made?
KNN methods
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
N/A
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
yes, understanding all the code about gridsearchCV. google it.
## How were these changes tested?
run the code
## Checklist
- [ X] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by Dmytro-Bonislavskyi — APPROVED** [review]
Well done!
Just a small comment regarding the evaluation:

Question 4: Model Evaluation
The task was to: "Using the best value for n_neighbors, fit a KNN model on the training data and **evaluate its performance on the test set** using accuracy_score."

The idea is to evaluate the model on the test data, which it did not see during the training process.
Currently, your code evaluates performance on the training data:

```
# Your code here...
wine_train['predicted']=knn.predict(wine_train.iloc[:, :-1])
wine_train_accuracy=accuracy_score(wine_train["class"],wine_train['predicted'])
wine_train_accuracy
```
